### PR TITLE
Internal refactoring and docs

### PR DESCRIPTION
**Pull request by zrzka**
* Internal refactoring
  * No breaking changes except one - `sys` module is private
* Forbids `unused_must_use` warning -> error
* Improves documentation
  * Only the part available at docs.rs
  * Internal documentation not touched

It's resurrected https://github.com/crossterm-rs/crossterm/pull/248.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,13 +1,446 @@
-#![deny(unused_imports)]
+#![deny(unused_imports, unused_must_use)]
 
+//! # Cursor
+//!
+//! The `crossterm_cursor` crate provides a functionality to work with the terminal cursor.
+//!
+//! This documentation does not contain lot of examples. The reason is that it's fairly
```
Comment (TimonPost) [inline src/lib.rs]:
```suggestion
//! This documentation does not contain a lot of examples. The reason is that it's fairly
```

File: src/lib.rs [hunk 2]
```diff
@@ -1,13 +1,446 @@
...
+/// Creates a new terminal cursor.
+pub fn cursor() -> TerminalCursor {
+    TerminalCursor::new()
+}
+
+/// A command to move the cursor to the given position.
+///
+/// # Notes
+///
+/// Commands must be executed / queued for execution otherwise they do nothing.
```
Comment (TimonPost) [inline src/lib.rs:205]: Would it be an idea to link to the book in those examples?

File: src/lib.rs [hunk 3]
```diff
@@ -1,13 +1,446 @@
...
+//! }
+//! ```
+//!
+//! Commands:
+//!
+//! ```no_run
+//! use std::io::{stdout, Write};
+//!
+//! use crossterm_cursor::{execute, Result};
+//! use crossterm_cursor::{BlinkOff, BlinkOn, Goto, ResetPos, SavePos};
```
Comment (TimonPost) [inline src/lib.rs]: can be inlined with the top

File: src/lib.rs (hunk 2 above)
Comment (zrzka) [reply src/lib.rs:205]: IIRC we agreed that the book documentation will be moved to the respective crates, mainly `lib.rs` and then the book will be completely removed until someone will have a time to write & maintain new book. Am I wrong?

File: src/lib.rs (hunk 1 above)
Comment (zrzka) [reply src/lib.rs]: Fixed.

File: src/lib.rs (hunk 3 above)
Comment (zrzka) [reply src/lib.rs]: Can you elaborate? Merge these two `use crossterm_cursor::...` or ...?

Comment (TimonPost) [reply src/lib.rs]: Yes exactly that

File: src/lib.rs (hunk 2 above)
Comment (TimonPost) [reply src/lib.rs:205]: You are correct. Don't mind this comment.

File: src/lib.rs (hunk 3 above)
Comment (zrzka) [reply src/lib.rs]: Okay, fixed.

**Review by TimonPost — APPROVED** [review]

**Pull request merged**